File: hw/cevallos/u1/hw011ChickenFarmv0.3.0/python/chicken_coop.py
from chicken import Chicken

class ChickenCoop:
    def __init__(self, coop_id: int, description: str):
        self.coop_id = coop_id
        self.description = description
        self.chickens = []  

    def add_chicken(self, chicken: Chicken):
        """Agrega una gallina al gallinero"""
        self.chickens.append(chicken)
        print(f"Chicken {chicken.name} added to coop {self.coop_id}")

    def remove_chicken(self, chicken_id: int):
        """Elimina una gallina por ID"""
        for chicken in self.chickens:
            if chicken.id == chicken_id:
                self.chickens.remove(chicken)
                print(f"Chicken with ID {chicken_id} removed from coop {self.coop_id}")
                return
        print(f"No chicken found with ID {chicken_id}")

    def list_chickens(self):
        """Muestra todas las gallinas del gallinero"""
        print(f"\nChicken Coop {self.coop_id} - {self.description}")
        if not self.chickens:
            print("No chickens in this coop.")
        else:
            for chicken in self.chickens:
                print(chicken)

    def make_all_do_stuff(self):
        """Hace que todas las gallinas hagan actividades aleatorias"""
        print(f"\n--- Chickens in Coop {self.coop_id} are active ---")
        for chicken in self.chickens:
            chicken.do_stuff()

    def count_chickens(self):
        """Devuelve la cantidad de gallinas en el gallinero"""
        return len(self.chickens)

    def __str__(self):
        return f"ChickenCoop{{id={self.coop_id}, description={self.description}, chickens={len(self.chickens)}}}"

File: hw/aymacana/u1/hw11ChickenCoops/ChickenCoopJS/chickenCoop.js
import { Chicken } from './chicken.js';

export class ChickenCoop {
    constructor(name = "Unnamed Coop") {
        this._chickens = [];
        this._name = name;
    }

    addChicken(chicken) {
        this._chickens.push(chicken);
        console.log(`Chicken '${chicken.name}' added to the coop '${this._name}'`);
    }

    removeChicken(chickenId) {
        for (let i = 0; i < this._chickens.length; i++) {
            if (this._chickens[i].id === chickenId) {
                const chicken = this._chickens[i];
                this._chickens.splice(i, 1);
                console.log(`Chicken '${chicken.name}' removed from the coop '${this._name}'`);
                return;
            }
        }
        console.log(`Chicken with ID ${chickenId} not found in coop '${this._name}'`);
    }

    getChicken(chickenId) {
        for (const chicken of this._chickens) {
            if (chicken.id === chickenId) {
                return chicken;
            }
        }
        return null;
    }

    getAllChickens() {
        return [...this._chickens];
    }

    countChickens() {
        return this._chickens.length;
    }

    isEmpty() {
        return this._chickens.length === 0;
    }

    displayChickens() {
        if (this.isEmpty()) {
            console.log(`Coop '${this._name}' is empty.`);
        } else {
            console.log(`\n--- Chickens in ${this._name} (${this.countChickens()} total) ---`);
            for (const chicken of this._chickens) {
                console.log(chicken.toString());
            }
        }
    }

    get name() {
        return this._name;
    }

    toString() {
        return `${this._name} with ${this.countChickens()} chickens`;
    }

    get length() {
        return this.countChickens();
    }
}